rflx/pyrflx/typevalue.py

```diff
@@ -514,11 +516,13 @@ def __init__(self, model: Message, refinements: Sequence[Refinement] = None) ->
             f.typeval.literals for f in self._fields.values() if isinstance(f.typeval, EnumValue)
         ]:
             self.__type_literals.update(t)
-        initial = self.Field(OpaqueValue(Opaque()))
+        self.__type_literals_keys = set(self.__type_literals)
+        initial = self.Field(OpaqueValue(Opaque()), INITIAL.name)
         initial.first = Number(0)
         initial.typeval.assign(bytes())
         self._fields[INITIAL.name] = initial
         self._simplified_mapping: Mapping[Name, Expr] = {}
+        self._simplified_mapping_keys: Set[Name] = set()
```

Comment:
Just curious: In which cases is it faster to have a separate set for the keys of a `dict`? Is there a rule of thumb?

Reply:
Set lookups are faster than dict key lookups. This is stated [here](https://stackoverflow.com/a/40963434/4790678) and the performance impact was also measurable in our case. It is especially useful  since we have many `in` checks, often without performing an operation on the dict.